src/sparsearraydok.jl

```diff
@@ -16,89 +18,46 @@ optionally with a function of type `F` to instantiate non-stored elements.
 struct SparseArrayDOK{T,N,F} <: AbstractSparseArray{T,N}
   storage::DOKStorage{T,N}
   size::NTuple{N,Int}
-  getunstoredindex::F
-
-  # bare constructor
-  function SparseArrayDOK{T,N,F}(
-    ::UndefInitializer, size::Dims{N}, getunstoredindex::F
-  ) where {T,N,F}
-    storage = DOKStorage{T,N}()
-    return new{T,N,F}(storage, size, getunstoredindex)
-  end
-
-  # unchecked constructor from data
-  function SparseArrayDOK{T,N,F}(
-    storage::DOKStorage{T,N}, size::Dims{N}, getunstoredindex::F
+  getunstored::F
+  global @inline function _SparseArrayDOK(
+    storage::DOKStorage{T,N}, size::Dims{N}, getunstored::F
   ) where {T,N,F}
-    return new{T,N,F}(storage, size, getunstoredindex)
+    return new{T,N,F}(storage, size, getunstored)
   end
 end
 
 # Constructors
 # ------------
 """
-    SparseArrayDOK{T}(undef, dims, unstored...)
-    SparseArrayDOK{T}(undef, dims...)
-    SparseArrayDOK{T,N}(undef, dims, unstored...)
-    SparseArrayDOK{T,N}(undef, dims...)
+    SparseArrayDOK{T}(undef, dims...[; getunstored])
+    SparseArrayDOK{T,N}(undef, dims...[; getunstored])
 
 Construct an uninitialized `N`-dimensional [`SparseArrayDOK`](@ref) containing
 elements of type `T`. `N` can either be supplied explicitly, or be determined by
 the length or number of `dims`.
 """
-SparseArrayDOK{T,N}(::UndefInitializer, dims, unstored...)
+SparseArrayDOK{T,N}(::UndefInitializer, dims; kwargs...)
 
 function SparseArrayDOK{T,N}(
```

Comment:
I think I would consider the final `uninit` constructor to also be an inner constructor, also because it somewhat hardcodes the storage type etc. Mostly just a preference though, fine with either.

Reply:
I think that makes sense, I'll change that.